src/parser/SHOGunApplicationUtil.ts

```diff
@@ -696,6 +694,62 @@ class SHOGunApplicationUtil<
     return xyzLayer;
   }
 
+  async parseMapboxStyleLayer(layer: S, projection?: OlProjectionLike) {
+    const {
+      url,
+      useBearerToken,
+      resolutions
+    } = layer.sourceConfig || {};
+
+    const {
+      minResolution,
+      maxResolution,
+      opacity,
+    } = layer.clientConfig || {};
+
+    const mapBoxLayerGroup = new OlLayerGroup({
+      minResolution,
+      maxResolution,
+      opacity
+    });
+
+    try {
+      await applyMapboxStyle(
+        mapBoxLayerGroup,
+        url,
+        {
+          projection: (projection instanceof OlProjection) ? projection.getCode() : projection,
+          resolutions: resolutions,
+          transformRequest: (resourceUrl) => {
+            return new Request(resourceUrl, {
+              headers: useBearerToken ? {
+                ...getBearerTokenHeader(this.client?.getKeycloak())
+              } : {}
+            });
+          }
+        }
+      );
+    } catch (e) {
+      Logger.error(`Could apply mapbox style to OlLayerGroup: ${e}`);
+    }
+
+    this.setLayerProperties(mapBoxLayerGroup, layer);
+
+    return mapBoxLayerGroup;
+  }
+
+  private forEachLayer(groupLayer: OlLayerGroup, callback: (layer: OlLayer) => void) {
+    groupLayer.getLayers().forEach(childLayer => {
+      if (childLayer instanceof OlLayerGroup) {
+        this.forEachLayer(childLayer, callback);
+      }
+
+      if (childLayer instanceof OlLayer) {
+        callback(childLayer);
+      }
+    });
+  }
+
```

Comment:
what is this function used for?

Reply:
Good catch, this function is currently not used and can be removed.

Reply:
see [4c4e4d6](https://github.com/terrestris/shogun-util/pull/869/commits/4c4e4d63e33c3593d9aeb82e2c2e6247f8c0b602)